pallets/providers/src/types.rs

```diff
@@ -0,0 +1,67 @@
+use codec::{Decode, Encode, MaxEncodedLen};
+use frame_support::pallet_prelude::*;
+use frame_support::traits::fungible::Inspect;
+use frame_support::BoundedVec;
+use scale_info::TypeInfo;
+
+use crate::Config;
+
+/// Structure that has the possible value propositions that a main storage provider can offer (and the runtime is aware of)
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct ValueProposition<T: Config> {
```

Comment:
We should add an `identifier` here so that when a bucket is assigned to a value proposition, it also enables off chain related business logic an MSP can execute locally or even provide additional off chain related fields associated to it. 

Having no identifier here is difficult to expand use cases and be generic over MSP solutions in the long run.

Reply:
Yes @ffarall and I discussed this earlier, it will be much easier to manage if the MSPs cannot modify existing value propositions as to not affect the current buckets associated to them. If they want they can create new value propositions entirely and users can modify their bucket to use the new one.